azurerm/internal/services/frontdoor/frontdoor_resource.go

```diff
@@ -423,17 +427,21 @@ func resourceArmFrontDoor() *schema.Resource {
 							Default:  0,
 						},
 						"custom_https_provisioning_enabled": {
-							Type:     schema.TypeBool,
-							Required: true,
+							Type:       schema.TypeBool,
+							Optional:   true,
+							Computed:   true,
+							Deprecated: "Deprecated in favour of `custom_https_configuration` resource",
```

Comment:
https://github.com/terraform-providers/terraform-provider-azurerm/pull/7498/files#r448222425
```suggestion
							Deprecated: "Deprecated in favour of `azurerm_frontdoor_custom_https_configuration` resource",
```